sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,476 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2024 ScyllaDB
+
+import json
+import logging
+import random
+import time
+from string import ascii_letters
+from threading import Event, Thread
+
+from confluent_kafka import Producer, Message
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from confluent_kafka.serialization import SerializationContext, MessageField, StringSerializer, StringDeserializer
+from confluent_kafka.schema_registry.avro import AvroSerializer, AvroDeserializer
+
+from sdcm.sct_config import SCTConfiguration
+from sdcm.sct_events import Severity
+from sdcm.sct_events.system import TestFrameworkEvent
+from sdcm.wait import wait_for
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def filter_kafka_options(**options: dict) -> dict:
+    """
+    Filters a dictionary of possible python thread options to keep only known Kafka options
+
+    :param options: dict, a dictionary of all options of a python thread
+
+    :return: dict, the known Kafka options.
+    """
+    known_options = {
+        'broker-list',
+        'bootstrap-server',
+        'topic',
+        'property',
+        'key-schema',
+        'value-schema',
+        'schema-registry',
+        'producer-property',
+        'consumer-property',
+        'compression-codec',
+        'batch-size',
+        'linger-ms',
+        'acks',
+        'key-separator',
+        'parse.key',
+    }
+
+    filtered_options = {}
+    for option, value in options.items():
+        if option in known_options:
+            # Handle nested options
+            if option in ['property', 'producer-property', 'consumer-property']:
+                if isinstance(value, dict):
+                    filtered_options[option] = value
+                elif isinstance(value, str) and '=' in value:
+                    sub_option, sub_value = value.split('=', 1)
+                    filtered_options.setdefault(option, {})[sub_option] = sub_value
+                else:
+                    LOGGER.error("Invalid format or type for %s: %s", option, value)
+            else:
+                filtered_options[option] = value
+
+    return filtered_options
+
+
+class KafkaThreadBase(Thread):
+    def __init__(self, tester, params: SCTConfiguration, timeout: int, **kwargs):
+        super().__init__(daemon=True)
+        self._check_thread_options(**kwargs)
```

Comment:
another approach would be to use `pop` instead `get` from `kwargs` and verify at the end that `kwargs` is empty. So we wouldn't have to force users using it and use defaults.